Source/Renderer/ShaderProgram.js

```diff
@@ -27,7 +29,51 @@ define([
     /**
      * @private
      */
+
+    function extractUniforms(shaderText) {
+        // TODO: regex doesn't work for inline structs
```

Comment:
We generally don't merge code with `TODO`s in them - is this hard to support?  If you need help, ask @tfili.

FYI, I don't think we use inline structs anywhere.  For example, here are the [built-in structs](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Source/Shaders/Builtin/Structs).